<launch>
	<!-- Data parameters -->
	<arg name="output_data_path" default="/home/jfgf/lolo_ws/src/algalbloom-tracking/output"/>
	<arg name="robot_name" default="lolo"/>
	

	<!-- Tracker node -->
	<group ns="$(arg robot_name)">
		<node name="algalbloom_tracker" pkg="smarc_algal_bloom_tracking" type="sam_gp4aes_controller.py" output="screen">
			<rosparam file="$(find smarc_algal_bloom_tracking)/config/controller_params.yaml"></rosparam>
			<param name="output_data_path" value="$(arg output_data_path)"/>
			
			<!-- Topics Subscribed -->
			<remap from="~measurement" to="algae_tracking/measurement"/>
			<remap from="~gps" to="core/gps" />
			<remap from="~gps_offset" to="algae_tracking/lat_lon_offset" />

			<!-- Topics Published -->
			<remap from="~gradient" to="algae_tracking/gradient"/>
			<remap from="~virtual_position" to="algae_tracking/vp"/>
			<remap from="~yaw_setpoint" to="ctrl/yaw_setpoint"/>
			<remap from="~thruster_1" to="core/thruster1_cmd"/>
			<remap from="~thruster_2" to="core/thruster2_cmd"/>
			
			<remap from="~live_waypoint" to="ctrl/waypoint_viz"/>

			<!-- Service Clients -->
			<remap from="~lat_lon_to_utm" to="lat_lon_to_utm"/> <!-- add dr/ for SAM -->

		</node>
	</group>

</launch>